pkg/build/api/types.go

```diff
@@ -352,6 +356,25 @@ type SourceBuildStrategy struct {
 	ForcePull bool
 }
 
+// BuildHookSpec holds build hooks with actions grouped by build status.
+type BuildHookSpec struct {
+	// The OnSuccess group is a list of BuildHooks that will be executed
+	// when the build changes to BuildPhaseComplete.
+	OnSuccess []BuildHook
+
+	// The OnFailure group is a list of BuildHooks that will be executed
+	// when the build changes to BuildPhaseFailed.
+	OnFailure []BuildHook
+}
+
+// A BuildHook defines actions to be executed when a build changes to a certain
+// phase.
+type BuildHook struct {
+	// StartBuilds holds a list of references to BuildConfigs that will be
+	// started when the hook is executed.
+	StartBuilds []kapi.ObjectReference
+}
```

Comment:
We should not store the `result` in `BuildStep`, because it is invalid to change the `BuildSpec` after the build is created. We'd need some new field in `Build.Status` for that.